src/RendererDX11.cpp

```diff
@@ -141,9 +145,96 @@ ID3D11ShaderResourceView* RendererDX11::get_render_texture_view() const
+    shadow_texture_desc.MipLevels = 1;
+    shadow_texture_desc.ArraySize = 1;
+    shadow_texture_desc.Format = DXGI_FORMAT_R24G8_TYPELESS;
+    shadow_texture_desc.SampleDesc.Count = 1;
+    shadow_texture_desc.SampleDesc.Quality = 0;
+    shadow_texture_desc.Usage = D3D11_USAGE_DEFAULT;
+    shadow_texture_desc.BindFlags = D3D11_BIND_DEPTH_STENCIL | D3D11_BIND_SHADER_RESOURCE;
+    shadow_texture_desc.CPUAccessFlags = 0;
+    shadow_texture_desc.MiscFlags = 0;
+
+    HRESULT hr = get_device()->CreateTexture2D(&shadow_texture_desc, nullptr, &m_instance_dx11->shadow_texture);
+    assert(SUCCEEDED(hr));
+
+    D3D11_DEPTH_STENCIL_VIEW_DESC shadow_depth_stencil_view_desc = {};
+    shadow_depth_stencil_view_desc.Format = DXGI_FORMAT_D24_UNORM_S8_UINT;
+    shadow_depth_stencil_view_desc.ViewDimension = D3D11_DSV_DIMENSION_TEXTURE2D;
+    shadow_depth_stencil_view_desc.Texture2D.MipSlice = 0;
+    shadow_depth_stencil_view_desc.Flags = 0;
+    hr = get_device()->CreateDepthStencilView(shadow_texture, &shadow_depth_stencil_view_desc, &m_instance_dx11->m_shadow_depth_stencil_view);
+    assert(SUCCEEDED(hr));
+
+    D3D11_SHADER_RESOURCE_VIEW_DESC shadow_shader_resource_view_desc = {};
+    shadow_shader_resource_view_desc.Format = DXGI_FORMAT_R24_UNORM_X8_TYPELESS;
+    shadow_shader_resource_view_desc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURE2D;
+    shadow_shader_resource_view_desc.Texture2D.MostDetailedMip = 0;
+    shadow_shader_resource_view_desc.Texture2D.MipLevels = shadow_texture_desc.MipLevels;
+    hr = get_device()->CreateShaderResourceView(shadow_texture, &shadow_shader_resource_view_desc, &m_instance_dx11->m_shadow_shader_resource_view);
+    assert(SUCCEEDED(hr));
+
+    D3D11_SAMPLER_DESC shadow_sampler_desc = {};
+    shadow_sampler_desc.Filter = D3D11_FILTER_COMPARISON_MIN_MAG_MIP_LINEAR;
+    shadow_sampler_desc.AddressU = D3D11_TEXTURE_ADDRESS_BORDER;
+    shadow_sampler_desc.AddressV = D3D11_TEXTURE_ADDRESS_BORDER;
+    shadow_sampler_desc.AddressW = D3D11_TEXTURE_ADDRESS_BORDER;
+    shadow_sampler_desc.MipLODBias = 0.0f;
+    shadow_sampler_desc.MaxAnisotropy = 1;
+    shadow_sampler_desc.ComparisonFunc = D3D11_COMPARISON_LESS_EQUAL;
+    shadow_sampler_desc.BorderColor[0] = 1.0f;
+    shadow_sampler_desc.BorderColor[1] = 1.0f;
+    shadow_sampler_desc.BorderColor[2] = 1.0f;
+    shadow_sampler_desc.BorderColor[3] = 1.0f;
+    shadow_sampler_desc.MinLOD = 0;
+    shadow_sampler_desc.MaxLOD = FLT_MAX;
+    hr = get_device()->CreateSamplerState(&shadow_sampler_desc, &m_instance_dx11->m_shadow_sampler_state);
+    assert(SUCCEEDED(hr));
+
+}
+
+void RendererDX11::bind_dsv_for_shadow_mapping() const
+{
+    auto const viewport = create_viewport(shadow_map_size, shadow_map_size);
+    get_device_context()->RSSetViewports(1, &viewport);
+    ID3D11RenderTargetView* renderTargets[] = {0};
+    get_device_context()->OMSetRenderTargets(1, renderTargets, m_instance_dx11->m_shadow_depth_stencil_view);
```

Comment:
This is kind of weird. We are creating RenderTarget each time this function is called, and never release previous render targets. We should probably just create one render target for shadow mapping or something, I guess...